plaso/storage/redis/redis_store.py

```diff
@@ -0,0 +1,489 @@
+    self._redis_client = None
+    self.serialization_format = definitions.SERIALIZER_FORMAT_JSON
+
+  @classmethod
+  def _SetClientName(cls, redis_client, name):
+    """Attempts to sets a Redis client name.
+
+    This method ignores errors from the Redis server or exceptions
+    indicating the method is missing, as setting the name is not a critical
+    function, and it is not currently supported by the fakeredis test library.
+
+    Args:
+      redis_client (Redis): an open Redis client.
+      name (str): name to set.
+    """
+    try:
+      redis_client.client_setname(name)
+    except redis.ResponseError as exception:
+      logger.debug(
+          'Unable to set redis client name: {0:s} with error: {1!s}'.format(
+              name, exception))
+
+  def _AddAttributeContainer(
+      self, container_type, container, serialized_data=None):
+    """Adds an attribute container to the store.
+
+   Args:
+      container_type (Type): container type attribute of the container being
+          added.
+      container (AttributeContainer): unserialized attribute container.
+      serialized_data (Optional[bytes]): serialized form of the container.
+    """
+    self._RaiseIfNotWritable()
+
+    identifier = identifiers.RedisKeyIdentifier()
+    container.SetIdentifier(identifier)
+
+    if not serialized_data:
+      serialized_data = self._SerializeAttributeContainer(container)
+
+    container_key = self._GenerateRedisKey(container_type)
+    string_identifier = identifier.CopyToString()
+    self._redis_client.hset(container_key, string_identifier, serialized_data)
+
+  def _GenerateRedisKey(self, subkey):
+    """Generates a Redis key inside the appropriate namespace.
+
+    Args:
+      subkey (str): Redis key to be prefixed with the namespace value.
+
+    Returns:
+      str: a Redis key name.
+    """
+    return '{0:s}-{1:s}-{2:s}'.format(
+        self._session_identifier, self._task_identifier, subkey)
+
+  def _GetAttributeContainers(self, container_type):
+    """Yields attribute containers
+
+    Args:
+      container_type (str): container type attribute of the container being
+          added.
+
+    Yields:
+      AttributeContainer: attribute container.
+    """
+    container_key = self._GenerateRedisKey(container_type)
+    for identifier, serialized_data in self._redis_client.hscan_iter(
+        container_key):
+      attribute_container = self._DeserializeAttributeContainer(
+          container_type, serialized_data)
+
+      identifier_string = identifier.decode('utf-8')
+      redis_identifier = identifiers.RedisKeyIdentifier(identifier_string)
+      attribute_container.SetIdentifier(redis_identifier)
+      yield attribute_container
+
+  def _GetAttributeContainerByIdentifier(self, container_type, identifier):
+    """Retrieves the container with a specific identifier.
+
+    Args:
+      container_type (str): container type.
+      identifier (RedisKeyIdentifier): event data identifier.
```

Comment:
`event data` => attribute container 

Since this seems to be a more generic use case than just event data.

Reply:
Done